### ISSUE TITLE
What is the intended use case?

### ISSUE THREAD

**Issue by preslavrachev** [Reading-Borough-Council/RBCInstaSearch#1]
I might be building a similar thing, so this seems very interesting.

**Comment by magicmilo** [comment]
The use case is to return comprehensive search results from thousands of articles and media as the user types with sub 50ms response times (instant) whilst being very low load and handling millions of request a second.

There's a lot of optimisation to do yet and also It needs to be configured to use generic input data.

What is your project?
I'd love some input.

If I'm honest it's mostly for fun and a challenge at the moment but the goal is for it to be the best and simplest to use web search around.

**Comment by magicmilo** [comment]
Also I'd be happy to modify it for your use case. It is currently being used for a as yet unfinished local government website.

**Comment by preslavrachev** [comment]
Thanks for the elaborate explanation. So, it is exactly what I was trying to build. At the moment, I am running a Go application called NoEmbed (https://noembed.preslav.me). The initial premise was to provide users with the ability to get nice paste-able snippets for every Web site they send me a URL for. With time, I added some comprehensive content extraction, since the majority of the available Web sites are in no way compliant with current metadata standards. 

I thought that as a nice extension to that, I could also index the content behind the provided URLs, and return it to others as some form of "related pages" or something.

So, in short, we can certainly combine forces on a common idea in the future.

**Comment by magicmilo** [comment]
Cool, looks great. That'd be great as well.
